tests/s3/versioning/test_versioning_multipart.py

```diff
@@ -0,0 +1,327 @@
+        self.log.info("Step 6: List Object Versions")
+        _, list_resp = check_list_object_versions(self.s3_ver_test_obj,
+                                                  bucket_name=self.bucket_name,
+                                                  expected_versions={})
+        versions = list_resp[1]["Versions"][0]
+        self.log.info("list_resp is %s",versions)
+        self.log.info("Step 7: Check GET/HEAD Object")
+        check_get_head_object_version(self.s3_test_obj, self.s3_ver_test_obj,
+                                      bucket_name=self.bucket_name, object_name=self.object_name,
+                                      version_id=versions["VersionId"],
+                                      etag=versions["ETag"])
+        self.log.info("Step 8: Check GET/HEAD Object with versionId = null")
+        check_get_head_object_version(self.s3_test_obj, self.s3_ver_test_obj,
+                                      bucket_name=self.bucket_name, object_name=self.object_name,
+                                      version_id="null", etag=versions["ETag"])
+        self.log.info("ENDED: Test pre-existing multipart upload in a versioning enabled bucket")
+
+    @pytest.mark.s3_ops
+    @pytest.mark.tags("TEST-41285")
+    @CTFailOn(error_handler)
+    def test_preexist_mpu_versioning_enabled_bkt_41285(self):
+        """
+        Test multipart upload in a versioning enabled bucket.
+        """
+        self.log.info("STARTED: Test multipart upload in a versioning enabled bucket.")
+        mp_config = MPART_CFG["test_40265"]
+        if os.path.exists(self.file_path):
+            os.remove(self.file_path)
+        create_file(self.file_path, mp_config["file_size"])
+        self.log.info("Step 2: PUT Bucket versioning with status as Enabled")
+        res = self.s3_ver_test_obj.put_bucket_versioning(bucket_name=self.bucket_name,
+                                                         status="Enabled")
+        assert_utils.assert_true(res[0], res[1])
+        precalc_parts = s3_utils.get_precalculated_parts(self.file_path, mp_config["part_sizes"],
+                                                         chunk_size=mp_config["chunk_size"])
+        self.log.info("Step 3-5: Upload multipart object to a bucket")
+        mpu_id, resp, sorted_lst = self.initiate_upload_list_mpu(self.bucket_name, self.object_name,
+                                                                 is_part_upload=True,
+                                                                 is_lst_mpu=True,
+                                                                 parts=precalc_parts)
+        self.log.info("Step 6: List multipart uploads")
+        res = self.s3_mp_test_obj.list_multipart_uploads(self.bucket_name)
+        assert_utils.assert_not_in("VersionId", res[1])
+        self.log.info("Step 7: Complete multipart upload")
+        _, resp = self.s3_mp_test_obj.complete_multipart_upload(mpu_id, sorted_lst,
```

Comment:
Please update lines 197 - 199 to call the updated upload_version as per comments added for TEST-41284